src/filter_finder.py

```diff
@@ -0,0 +1,165 @@
+
+As of 04/22/2019: In order to use this script properly, the values for path, lower_chosen, and upper_chosen must be manually edited into this code.
+
+TODO: make this script configurable to run from the commandline with no need to edit code, e.g. through `python board_filter -v "board.mp4" -c "wheels"`
+
+This script was created as part of a school project, and therefore license-free.
+"""
+
+import os.path
+import cv2
+import numpy as np
+
+__author__ = "Nick"
+
+
+#Increments by which values can be changed through keyboard controls
+sinc = 1
+binc = 5
+
+path = 'We have to manually enter the path to the video file here until we add support for commandline parameters.' #TODO: make path passable from the commandline
+
+#if path leads to valid file
+if os.path.isfile(path):
+	cap = cv2.VideoCapture(path)
+	
+	#try to read first frame
+	success, frame = cap.read()
+	
+	if(success):
+		#get dimensions of this video
+		height, width, _ = frame.shape
+
+		#lower and upper bounds for color detections, to be derived using keyboard controls. The default values given here are the averages of previous trials with Steve's skateboard.
+		lower_board = np.array([82,39,80])
+		upper_board = np.array([100,155,180])
+
+		lower_wheel = np.array([10,40,0])
+		upper_wheel = np.array([29,230,255])
+
+		lower_edge = np.array([10,25,115])
+		upper_edge = np.array([25,155,255])
+
+		#decide which color range to experiment on during this run
+		lower_chosen = lower_edge
+		upper_chosen = upper_edge
+
+		#factor by which to downsample each frame
+		scale = 2
+
+		#go through all the frames of this video
+		while(success):
+			#rescale frame for faster processing
+			frame = cv2.resize(frame, (width / scale, height / scale))
+			
+			#convert frame to HSV for easier processing
+			hsv_frame = cv2.cvtColor(frame, cv2.COLOR_BGR2HSV)
+			
+			#get masks for individual board features
+			board_mask = cv2.inRange(hsv_frame, lower_board, upper_board)
+			wheel_mask = cv2.inRange(hsv_frame, lower_wheel, upper_wheel)
+			edge_mask = cv2.inRange(hsv_frame, lower_edge, upper_edge)
+			
+			#combine feature masks into one
+			skateboard_mask = cv2.bitwise_or(board_mask, wheel_mask)
+			skateboard_mask = cv2.bitwise_or(skateboard_mask, edge_mask)
+			
+			#Extract features from frame using masks
+			extracted_board = cv2.bitwise_and(frame, frame, mask = board_mask)
+			extracted_wheel = cv2.bitwise_and(frame, frame, mask = wheel_mask)
+			extracted_edge = cv2.bitwise_and(frame, frame, mask = edge_mask)
+			extracted_skateboard = cv2.bitwise_and(frame, frame, mask = skateboard_mask)
+			
+			#show raw frame
+			cv2.imshow('raw frame', frame)
+
+			#show all extracted features
+			cv2.imshow('extracted board', extracted_board)
+			cv2.imshow('extracted wheels', extracted_wheel)
+			cv2.imshow('extracted edges', extracted_edge)
+			cv2.imshow('extracted skateboard', extracted_skateboard)
+			
+			#exit early if escape key pressed
```

Comment:
the code is clear enough, comments should generally be *why* not *what*

Reply:
But the clarity and readability isn't decreased by this comment